File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
const eraseOverlapIntervals = intervals => {
    intervals.sort((a, b) => a[0] - b[0]);
    let count = 0, prevEnd = intervals[0][1];
    for (let i = 1; i < intervals.length; i++) {
        if (intervals[i][0] < prevEnd) {
            count++;
            prevEnd = Math.min(intervals[i][1], prevEnd);
        } else {
            prevEnd = intervals[i][1];
        }
    }
    return count;
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        count = 0
        def sortBy(a):
            return a[0]
        intervals.sort(key=sortBy)
        prev_endpoint = intervals[0][1]
        for interval in intervals[1:]:
            if interval[0] < prev_endpoint:
                prev_endpoint = min(prev_endpoint, interval[1])
                count += 1
            else:
                prev_endpoint = interval[1]
        return count